src/promptflow/promptflow/_utils/version_hint_utils.py

```diff
@@ -0,0 +1,117 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+import datetime
+import json
+import sys
+import contextlib
+
+from promptflow._constants import (LAST_HINT_TIME, LAST_CHECK_TIME, PF_VERSION_CHECK, CLI_PACKAGE_NAME,
+                                   HINT_FREQUENCY_DAY, GET_PYPI_FREQUENCY_DAY, LATEST_VERSION, CURRENT_VERSION)
+from promptflow._sdk._constants import HOME_PROMPT_FLOW_DIR
+
+
+if sys.platform.startswith("win"):
+    import msvcrt
+else:
+    import fcntl
+
+HINT_ACTIVITY_NAME = ["pf.flows.test", "pf.runs.create_or_update", "pfazure.flows.create_or_update",
```

Comment:
show we also do hint for actions that no hurt like: 
pf
pf --version
pf --help

Reply:
I think this is just for CLI?

Reply:
offline sync, no need ad for now